consensus/src/simplex/signing_scheme/ed25519.rs

```diff
@@ -8,50 +8,79 @@
 use crate::{
     simplex::{
         signing_scheme::{self, utils::Signers, vote_namespace_and_message},
-        types::{Participants, Vote, VoteContext, VoteVerification},
+        types::{OrderedExt, Vote, VoteContext, VoteVerification},
     },
     types::Round,
 };
 use bytes::{Buf, BufMut};
 use commonware_codec::{EncodeSize, Error, Read, ReadRangeExt, Write};
 use commonware_cryptography::{
-    ed25519::{Batch, PrivateKey, PublicKey, Signature as Ed25519Signature},
-    BatchVerifier, Digest, Signer as _, Verifier as _,
+    ed25519::{self, Batch},
+    BatchVerifier, Digest, PublicKey, Signer as _, Verifier as _,
 };
+use commonware_utils::set::Ordered;
 use rand::{CryptoRng, Rng};
 use std::collections::BTreeSet;
 
 /// Ed25519 implementation of the [`Scheme`] trait.
 #[derive(Clone, Debug)]
-pub struct Scheme {
-    /// Participant set (sorted) used for signer indexing and batch verification.
-    participants: Participants<PublicKey>,
-    /// Optional local signing key paired with its participant index.
-    signer: Option<(u32, PrivateKey)>,
+pub struct Scheme<P: PublicKey> {
+    /// Participant set's public identities.
+    participants: Ordered<P>,
+    /// Participant set used for consensus signer indexing and verification.
+    consensus: Vec<ed25519::PublicKey>,
+    /// Optional local consensus signing key paired with its participant index.
+    signer: Option<(u32, ed25519::PrivateKey)>,
 }
 
-impl Scheme {
+impl<P: PublicKey> Scheme<P> {
     /// Creates a new scheme instance with the provided key material.
     ///
-    /// * `participants` - ordered validator set used for verification.
-    /// * `private_key` - optional secret key enabling signing capabilities.
-    pub fn new(participants: impl Into<Participants<PublicKey>>, private_key: PrivateKey) -> Self {
-        let participants = participants.into();
-
-        let signer = participants
-            .index(&private_key.public_key())
-            .map(|index| (index, private_key));
+    /// Participants are provided as tuples pairing identity keys with consensus keys.
+    /// The identity key (first element) is used for committee ordering and indexing,
+    /// while the consensus key (second element) is used for signing and verification.
+    ///
+    /// If the provided private key does not match any consensus key in the participant set,
+    /// the instance will act as a verifier (unable to sign votes).
+    pub fn new(
+        mut participants: Vec<(P, ed25519::PublicKey)>,
```

Comment:
We can still provide `Ordered` here (and just `sort_by` on creation like we do in `p2p`)?

Reply:
This would require them to sort correctly (we wouldn't sort again here): https://github.com/commonwarexyz/monorepo/blob/778d3fe37222044912ba6d05d6eb9d7610268ef9/p2p/src/authenticated/lookup/actors/tracker/actor.rs#L302